nuxeo-core/nuxeo-core-event/src/main/java/org/nuxeo/ecm/core/work/StreamWorkManager.java

```diff
@@ -410,6 +388,16 @@ public boolean shutdown(long timeout, TimeUnit timeUnit) {
         shutdownInProgress = true;
         try {
             boolean ret = streamProcessor.stop(Duration.ofMillis(timeUnit.toMillis(timeout)));
+            // sleep for a given amount of time for works to have time to persist their state and stop properly
```

Comment:
simply change set the timeout duration of the streamProcessor.stop to max(timeout, shutdownDelay)